### PR TITLE
CheckTranslationReference lint pass learns additional checks

### PR DESCRIPTION
**Pull request by RoosterDragon**
- Test all translation languages, not just English.
- Report any fields marked with TranslationReferenceAttribute that the lint pass lacked the knowledge to check.
- Improve context provided by lint messages.
- Restructure code for readability.

Additionally:
- Fix some bugs in Translation.
- Tweak some UI Yaml files.
- Inject platform dependency in SpriteFont.

Assists with #10475. Split from #21293

### CODE REVIEW

**Review by PunkPun — COMMENTED** [review]
this is quite wordy 😅 

OpenRA.Utility(1,1): Warning: Missing attribute `description` of key `dropdown-map-creeps` in `en` language in mod translation files required by Actor `world` trait `MapCreeps.CheckboxDescription`

it also seems that there's a wording missmatch

Testing map: Atreides 01a
OpenRA.Utility(1,1): Warning: Missing key `dropdown-map-creeps.description` in `en` language in map translation files required by Actor `world` trait `MapCreeps.CheckboxDescription`


When I tried to overwrite a key in fort lonestar I got spammed with errors. There should be just 3
Screenshot 2024-04-06 at 12 55 59

File: OpenRA.Mods.Common/Lint/CheckTranslationReference.cs [hunk 1]
```diff
@@ -23,289 +24,359 @@ namespace OpenRA.Mods.Common.Lint
...
+
+			return chromeLayoutNodes;
+		}
 
-			foreach (var filename in modData.Manifest.ChromeLayout)
+		static void CheckChrome(
+			MiniYamlNode rootNode, Translation translation, string language,
+			Action<string> emitError, Action<string> emitWarning,
+			Dictionary<(string WidgetName, string FieldName), TranslationReferenceAttribute> translationReferencesByWidgetField,
+			TranslationKeys usedKeys)
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Lint/CheckTranslationReference.cs:204]: CheckChrome doesn't do anything with `Translation translation, string language, Action<string> emitError, Action<string> emitWarning`

Comment (RoosterDragon) [reply OpenRA.Mods.Common/Lint/CheckTranslationReference.cs:204]: Removed

**Comment by RoosterDragon** [comment]
> this is quite wordy 😅

Open to wording suggestions, I wanted to ensure the message provided plenty of detail so naturally it's not a short message.

> it also seems that there's a wording missmatch

Unified to "Missing key <key>" for both cases.

> When I tried to overwrite a key in fort lonestar I got spammed with errors. There should be just 3

Fixed.

Thanks for the review, this certainly tidies up a few rough edges.

**Review by PunkPun — APPROVED** [review]
LGETM

**Comment by PunkPun** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/20b314f409d93918f4796e6852f2c3b1ebfcf826...4763889cfdac86b304f8404962c0ee78dbd8462e)

**Pull request merged**